message_ix/cli.py

```diff
@@ -12,11 +12,13 @@
 from message_ix.default_path_constants import CONFIG_PATH, DEFAULT_MODEL_PATH
 from message_ix.utils import logger
 
+GMS_EXT = ['.gms', '.opt', '.md']
 
-def recursive_copy(src, dst, overwrite=False):
+
+def recursive_copy(src, dst, overwrite=False, valid_ext=GMS_EXT):
```

Comment:
ah, here you have added a white list instead of a black list. I would instead suggest changing this to something like

```
def recursive_copy(src, dst, overwrite=False, skip_ext=[]): # now a black list
```

And then wherever this function is being called, specifically set the blacklist

```
recursive_copy(src, dst, skip_ext=['gdx'])
```